### PR TITLE
Improve Making a New Learning Environment article

### PR DESCRIPTION
**Pull request by jo3w4rd**
The old example only trained efficiently if you you used "shaped" rewards. despite this, we removed the shaped rewards, so the example only trained in > 500,000 steps (if at all).

This update simplifies the reward structure so that it only sets a reward of 1 when the agent reaches the target. In general training is more successful in 0.5 (and fails to progress in 0.4), but still takes a long, long time. 

The massive improvement in training comes from setting the batch_size and buffer_size hyperparameters to small values. A higher DecisionFrequency on the Agent also helps. I added a training section to the article to cover these changes and to briefly review the training process.

### CODE REVIEW

**Comment by jo3w4rd** [comment]
@awjuliani This is specifically for v0.5. I need to do a different rewrite for v0.6.

**Comment by vincentpierre** [comment]
@jo3w4rd Since this is not a hotfix, can we make these changes for v0.6 ?

**Comment by jo3w4rd** [comment]
@vincentpierre Yes, I plan to make these and additional changes to v0.6. But I don't see why we wouldn't fix v0.5 along the way. Can we treat it like a hot fix?

**Comment by jo3w4rd** [comment]
@vincentpierre , can you approve this PR (or close it, if you don't want the improvements in the 0.5 docs).

**Comment by xiaomaogy** [comment]
Hi @jo3w4rd , I think folks are no longer interested in a fix in v0.5, can you please update this PR to be against the v0.6, and I will help with the PR review?

**Comment by jo3w4rd** [comment]
Already merged: https://github.com/Unity-Technologies/ml-agents/pull/1404
